Makefile

```diff
@@ -2,34 +2,40 @@
 SRC := src
 INCLUDE := include
 BIN := bin
+DIST := dist
 
 # Compile Vars
-CC := clang
-CFLAGS := -I$(INCLUDE) -Wall -Wextra -Werror -fopenmp
-LDFLAGS := -lomp -L/lib/llvm-14/lib
+CC := gcc
+CFLAGS := -I$(INCLUDE) -Wall -Wextra -Werror -O3
+LDFLAGS := -fopenmp -flto -lSDL2
 TARGET := $(BIN)/fractal
 
-all: clean $(TARGET)
+all: clean setup $(TARGET)
 
-$(TARGET): bmp.o main.o fractal.o
-	$(CC) $(LDFLAGS) $^ -o $@
+setup:
+	mkdir -p $(BIN) $(DIST)
 
-bmp.o:
+$(TARGET): $(DIST)/bmp.o $(DIST)/main.o $(DIST)/fractal.o 
+	$(CC) $(LDFLAGS) $^ -o $@ -lSDL2
+
+$(DIST)/bmp.o:
 	$(CC) $(CFLAGS) -c $(SRC)/bmp.c -o $@
 
-main.o:
-	$(CC) $(CFLAGS) -c $(SRC)/main.c -o $@
+$(DIST)/main.o:
+	$(CC) $(CFLAGS) -lSDL2 -c $(SRC)/main.c -o $@

```

Comment:
`-lSDL2` is a linker flag. It doesn't need to be added to any compile commands. The flag `-fopenmp` is an exception because it also tells the compiler where to find the openmp header (include flags are compiler flags).

A summary of compile flags vs link flags would be this:
Compile flags (CFLAGS) are all the flags required to compile .c files into .o files. Among others, they include:
1. Include paths (where to find .h files). `-Ipath/to/headers`
2. Optimization flags. `-O3`
3. Warning controlling flags. `-Wall -Werror -Wextra`
Linker flags (LDFLAGS) are all the flags required to link all .o files into a binary. Among others, they include:
1. Libraries. `-lSDL2`
2. Library paths (where to find the linked libraries). `-Lpath/to/lib`
3. Linker optimizations. `-flto`

Other "special" flags are linker and compiler flags, such as `-fopenmp`. It is a compile flag because it tells the compiler to find the OpenMP headers. It is a linker flag because it tells the compiler to link against the openmp library and where to find it.